src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs

```diff
@@ -187,36 +187,31 @@ public void TestMachineName()
             Assert.NotNull(_process.MachineName);
         }
 
-        [Fact]
+        [Fact, PlatformSpecific(~PlatformID.OSX)]
         public void TestMainModule()
         {
+            // Ensure the process has loaded the modules.
+            while (_process.Modules.Count == 0) ;
```

Comment:
Is there any chance this could infinitely loop? I guess the individual test method would timeout after a point even if it did, right? Adding an individual timeout for this might give a better error message, though.